modules/shortcodes/tweet.php

```diff
@@ -142,4 +143,61 @@ static public function jetpack_tweet_shortcode_script() {
 		}
 	}
 
+	static public function enqueue_block_editor_assets() {
+		wp_enqueue_script( 'wp-blocks' );
+		wp_enqueue_script( 'wp-i18n' );
+		wp_enqueue_script( 'wp-element' );
+		wp_enqueue_script( 'shortcode' );
+		add_action( 'admin_print_footer_scripts', array( __CLASS__, 'admin_footer' ), 999 );
+	}
+
+	static public function admin_footer() {
+?>
+<script>
+( function( wp ) {
+	var blockStyle = {
+			backgroundColor: '#900',
+			color: '#fff',
+			padding: '20px'
+		};
+
+	wp.blocks.registerBlockType( 'jetpack/tweet', {
+		title: wp.i18n.__( 'Tweet', 'jetpack' ),
+		icon: 'twitter',
+		category: 'layout',
+
+		attributes : {
+			tweet : wp.blocks.query.query( 'input[name=tweet]').value
```

Comment:
The idea with [attribute matchers](http://gutenberg-devdoc.surge.sh/reference/attribute-matchers/) (or, perhaps soon to be called ["sourced" values](https://github.com/WordPress/gutenberg/pull/1902)) is to avoid duplicating data between [comment attributes](https://github.com/WordPress/gutenberg/blob/master/docs/faq.md#how-is-data-stored-ive-seen-html-comments-what-is-their-purpose) and the actual saved markup of the block. For example, we don't need to serialize an image's URL into the comments blob if we know we can easily grab it from the `img`'s `src` attribute. In more complex examples, duplication is inevitable or there's a need for hidden metadata.

So what we define in `attributes` purely relates to _getting an object representation of the block_ back from the saved content, nothing to do with the `edit`-able render.

In your case, the saved content is a shortcode, from which you want to extract the `tweet` value. There's not a matcher for this (arguments could be made either way for whether we should [add conveniences for shortcodes](https://github.com/WordPress/gutenberg/issues/334)), so you might just want to let this live in comment metadata. Currently this is automatic if you set an attribute from your `edit` form but don't define it in the `attributes` property, but this is [likely to change soon](https://github.com/WordPress/gutenberg/pull/1905) to be more explicit (i.e. define all attributes, both the ones sourced from content and those serialized into comments).

I'm supposing you were expecting that defining this here would establish a link between the attribute value and the input from `edit`, which is not the case. Instead, you'll want to explicitly call `props.setAttributes` from your `edit` function when the input changes, similar to the concept of a [controlled input](https://facebook.github.io/react/docs/forms.html#controlled-components) in React.

```js
onInput: function( event ) {
	props.setAttributes({
		tweet: event.target.value
	});
}
```